### PR TITLE
fix(transferownershipmodal): remove linting errors

### PR DESCRIPTION
**Pull request by wanlingt**
## Problem

The linting warnings for TransferOwnershipModal are annoying and should be fixed.

## Solution

Fixed the warnings by adding the missing dependencies and removing forbidden non-null assertions

## Tests

**1. Regression test for transferring ownership of all forms**
- [x] Log into the FormSG dashboard
- [x] Transfer all forms to another admin
- [x] Sign in as the other admin and check that all forms from the previous admin have been transferred over

### CODE REVIEW

File: frontend/src/features/user/transfer-ownership/TransferOwnershipModal.tsx [hunk 1]
```diff
@@ -38,7 +43,22 @@ type TransferOwnershipInputs = {
...
+const useModalState = ({
+  onClose,
+  reset,
+  trigger,
+}: {
+  onClose: () => void
+  reset: UseFormReset<TransferOwnershipInputs>
+  trigger: UseFormTrigger<TransferOwnershipInputs>
+}): {
+  page: number
```
Comment (foochifa) [inline frontend/src/features/user/transfer-ownership/TransferOwnershipModal.tsx]: This is a lil hard to read should we refactor it into `useModalStateReturn` 

then 
```
}) : useModalStateReturn => {
```

File: frontend/src/features/user/transfer-ownership/TransferOwnershipModal.tsx [hunk 2]
```diff
@@ -80,16 +100,14 @@ const useModalState = ({ onClose, reset, trigger }) => {
         },
       },
     )
-  }, [page, email, user])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (foochifa) [inline frontend/src/features/user/transfer-ownership/TransferOwnershipModal.tsx]: Ooo is there a reason why we are disabling lint here for `refetch()` and `resetModal()` does it break anything 😅

File: frontend/src/features/public-form/PublicFormProvider.tsx [hunk 3]
```diff
@@ -676,11 +676,15 @@ export const PublicFormProvider = ({
       submitEmailModeFormFetchMutation,
       submitEmailModeFormMutation,
       enableEncryptionBoundaryShift,
+      enableVirusScanner,
+      submitStorageModeClearFormMutation,
```
Comment (foochifa) [inline frontend/src/features/public-form/PublicFormProvider.tsx:680]: Are these vals being used in the content of the provider?

**Review by foochifa — COMMENTED** [review]
Thanks for fixing the lints @wanlingt, just some comments!

File: frontend/src/features/public-form/PublicFormProvider.tsx (hunk 3 above)
Comment (wanlingt) [reply frontend/src/features/public-form/PublicFormProvider.tsx:680]: Yup! enableVirusScanner used in [L560](https://github.com/opengovsg/FormSG/blob/e9bd3fcfaf8998e1a841373f4c5c7a92b217b99e/frontend/src/features/public-form/PublicFormProvider.tsx#L560), submitStorageModeClearFormMutation in [L608](https://github.com/opengovsg/FormSG/blob/e9bd3fcfaf8998e1a841373f4c5c7a92b217b99e/frontend/src/features/public-form/PublicFormProvider.tsx#L608)

File: frontend/src/features/user/transfer-ownership/TransferOwnershipModal.tsx (hunk 1 above)
Comment (wanlingt) [reply frontend/src/features/user/transfer-ownership/TransferOwnershipModal.tsx]: Done :) [5d4e67b](https://github.com/opengovsg/FormSG/pull/6802/commits/5d4e67b193fd04f18910d7e00315a5bdab096531)

File: frontend/src/features/user/transfer-ownership/TransferOwnershipModal.tsx (hunk 2 above)
Comment (wanlingt) [reply frontend/src/features/user/transfer-ownership/TransferOwnershipModal.tsx]: Oops no reason, forgot to remove it heh. Amended here [2b2d46f](https://github.com/opengovsg/FormSG/pull/6802/commits/2b2d46ff47c768738cd678bdad4f3e7ad3fe7e9e)

**Review by foochifa — APPROVED** [review]
LGTM Thanks @wanlingt

**Pull request merged**